metaflow/plugins/argo/argo_workflows_decorator.py

```diff
@@ -61,3 +106,40 @@ def task_finished(
         # by the next task here.
         with open("/mnt/out/task_id", "w") as file:
             file.write(self.task_id)
+
+        # Emit Argo Events given that the flow has succeeded. Given that we only
+        # emit events when the task succeeds, we can piggy back on this decorator
+        # hook which is guaranteed to execute only after rest of the task has
+        # finished execution.
+
+        if self.attributes["auto-emit-argo-events"]:
+            # Event name is set to metaflow.project.branch.flow.step
+            # TODO: Check length limits for fields in Argo Events
+            event = ArgoEvent(
+                name="metaflow.%s.%s"
+                % (current.get("project_flow_name", flow.name), step_name)
+            )
+            # There should only be one event generated even when the task is retried.
+            event.add_to_payload("id", current.pathspec)
+            event.add_to_payload("pathspec", current.pathspec)
+            event.add_to_payload("flow_name", flow.name)
+            event.add_to_payload("run_id", self.run_id)
+            event.add_to_payload("step_name", step_name)
+            event.add_to_payload("task_id", self.task_id)
+            # Add @project decorator related fields. These are used to subset
+            # @trigger_on_finish related filters.
+            for key in (
+                "project_name",
+                "branch_name",
+                "is_user_branch",
+                "is_production",
+                "project_flow_name",
+            ):
+                if current.get(key):
+                    event.add_to_payload(key, current.get(key))
+            # Add more fields here...
+            event.add_to_payload("auto-generated-by-metaflow", True)
+            # Keep in mind that any errors raised here will fail the run but the task
+            # will still be marked as success. That's why we explicitly swallow any
+            # errors and instead print them to std.err.
+            event.publish(ignore_errors=True)
```

Comment:
Discussion only.

I guess this is a difficulty with distributed systems.  So publish fails here.  How does the user find out?

Event source flow owner won't know, everything is OK (Run was successful!).

Event target flow owner can only know when they realize the source flow ran fine, but the target flow did not get triggered.

That is to say events plumbing has to be 100% reliable (or supporting infra must have failsafe alerting mechanisms).